executors/keptn/pkg/keptn/keptn.go

```diff
@@ -0,0 +1,329 @@
+	}
+
+	apiHandler := apiutils.NewAuthenticatedAPIHandler(url, t, "x-token", nil, "http")
+	resourceHandler := apiutils.NewAuthenticatedResourceHandler(url, t, "x-token", nil, "http")
+	projectHandler := apiutils.NewAuthenticatedProjectHandler(url, t, "x-token", nil, "http")
+	eventHandler := apiutils.NewAuthenticatedEventHandler(url, t, "x-token", nil, "http")
+
+	return &Keptn{
+		url:             url,
+		token:           &t,
+		apiHandler:      apiHandler,
+		resourceHandler: resourceHandler,
+		projectHandler:  projectHandler,
+		eventHandler:    eventHandler,
+		git:             git,
+	}, nil
+}
+
+func (k *Keptn) CreateOrUpdateProject(project string, shipyard string) error {
+	encodedShipyardContent := base64.StdEncoding.EncodeToString([]byte(shipyard))
+	projectInfo := apimodels.CreateProject{
+		Name:     &project,
+		Shipyard: &encodedShipyardContent,
+	}
+
+	projectGetInfo := apimodels.Project{
+		ProjectName: project,
+	}
+
+	if k.git != nil {
+		projectInfo.GitRemoteURL = k.git.URL
+		projectInfo.GitToken = k.git.Token
+		projectInfo.GitUser = k.git.User
+
+		projectGetInfo.GitRemoteURI = k.git.URL
+		projectGetInfo.GitToken = k.git.Token
+		projectGetInfo.GitUser = k.git.User
+	}
+
+	if _, kErr := k.projectHandler.GetProject(apimodels.Project{
+		ProjectName:     project,
+		ShipyardVersion: shipyard,
+		Stages:          []*apimodels.Stage{},
+	}); kErr == nil {
+		fmt.Println("found the project - deleting it now")
+		if _, kErr := k.apiHandler.DeleteProject(projectGetInfo); kErr != nil {
+			return fmt.Errorf("failed to delete project with err: %v", kErr.GetMessage())
+		}
+	}
+
+	if _, kErr := k.apiHandler.CreateProject(projectInfo); kErr != nil {
+		return fmt.Errorf("failed to create project with err: %v", kErr.GetMessage())
+	}
+
+	return nil
+}
+
+func (k *Keptn) DeleteProject(project string) error {
+	p := apimodels.Project{
+		ProjectName: project,
+	}
+
+	if _, kErr := k.apiHandler.DeleteProject(p); kErr != nil {
+		return fmt.Errorf("%v : %w", kErr.GetMessage(), ErrFailedDeleteProject)
+	}
+
+	return nil
+}
+func (k *Keptn) CreateService(service, project string) error {
+	if _, kErr := k.apiHandler.CreateService(project, apimodels.CreateService{
+		ServiceName: &service,
+	}); kErr != nil {
+		return fmt.Errorf("failed to create service with err: %v", kErr.GetMessage())
+	}
+
+	return nil
+}
+
+func (k *Keptn) AddResourceToAllStages(service, project, resourceName, resourceContent string) error {
+	stages, err := k.getProjectStages(project)
+	if err != nil {
+		return err
+	}
+
+	for _, stage := range stages {
+		if err := k.AddResourceToStage(service, project, stage.StageName, resourceNameToURI(resourceName), resourceContent); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (k *Keptn) AddResourceToStage(service, project, stage, resourceURI, resourceContent string) error {
+	// encodedResourceContent := base64.StdEncoding.EncodeToString([]byte(resourceContent))
+	encodedResourceContent := resourceContent
```

Comment:
Why rename? This looks like it could be removed

Reply:
yea I think it was cruft left from the comment above. Will remove.